### PR TITLE
Isbm module run parambug 42270

### PR DESCRIPTION
**Pull request by isbm**
### What does this PR do?

Fixes the issue on `module.run`: https://github.com/saltstack/salt/pull/42270
Fixes the documentation for `module.run`. Allows running the same function within the same functionset with the different parameters. Fixes tests.

Note: probably needs to be backported to 2017.7

### Tests written?

Yes

### CODE REVIEW

**Comment by isbm** [comment]
@cachedout I think this is needed to go to the 2017.7 before the release...

**Comment by cachedout** [comment]
@isbm It won't make it into the release, sorry.

File: doc/topics/releases/2017.7.0.rst [hunk 1]
```diff
@@ -113,27 +113,57 @@ State Module Changes
...
-
   Since a lot of users are already using :py:func:`module.run
   <salt.states.module.run>` states, this new behavior must currently be
   explicitly turned on, to allow users to take their time updating their SLS
   files. However, please keep in mind that the new syntax will take effect in
   the next feature release of Salt (Oxygen) and the old usage will no longer be
   supported at that time.
 
+  Another feature of the new :py:func:`module.run <salt.states.module.run>` is that
+  it allows to call many functions in a single batch, such as:
```
Comment (cachedout) [inline doc/topics/releases/2017.7.0.rst]: `it allows calling many functions`...

File: doc/topics/releases/2017.7.0.rst [hunk 2]
```diff
@@ -113,27 +113,57 @@ State Module Changes
...
+  .. code-block:: yaml
+
+      run_something:
+        module.run:
+          - mymodule.function_without_parameters:
+          - mymodule.another_function:
+            - myparam
+            - my_other_param
+
+  In a rare case you have a function that needs to be called several times but
```
Comment (cachedout) [inline doc/topics/releases/2017.7.0.rst]: `In the rare case that you have`...

File: doc/topics/releases/2017.7.0.rst [hunk 3]
```diff
@@ -113,27 +113,57 @@ State Module Changes
...
+  with the different parameters, an additional feature of "tagging" is to the
+  rescue. In order to tag a function, use a colon delimeter. For example
```
Comment (cachedout) [inline doc/topics/releases/2017.7.0.rst]: `For example:`

**Comment by isbm** [comment]
@cachedout all these pytests aren't really related to this particular PR, right? Or I am missing something?

**Comment by rallytime** [comment]
Most of these have been fixed upstream, but I can update this branch just to be sure we don't miss anything. Thanks @isbm!

**Pull request closed without merging**